R/Plot_synergy.R

```diff
@@ -1,328 +1,327 @@
-                   cex = 0.8, z = list(tick.number = 6),
-                   x=xaxis, y = yaxis)
-      zlabs <- list(expression("Synergy score"), rot = 90, 
-                    cex = 1, axis.key.padding = 0)
-      xpar <- list(as.character(drug.col), cex = 1, rot = 20)
-      ypar <- list(as.character(drug.row), cex = 1, rot = -50)
-      
-      syn.3d.plot <- wireframe(t(mat.tmp),scales = par1,
-                       drape = TRUE, colorkey = list(space = "top",width = 0.5),
-                       screen = list(z = 30, x = -55),
-                       zlab = zlabs, xlab = xpar, ylab = ypar,
-                       zlim = c(start.point, end.point),
-                       col.regions = col,
-                       main = plot.title,
-                       at = do.breaks(c(start.point, end.point), length(col)),
-                       par.settings = list(axis.line=list(col="transparent")),
-                       zoom = 1, aspect = 1)
-      # 2d plot
-      layout(matrix(c(1, 2, 3, 3), nrow = 2L, ncol = 2L), heights = c(0.1, 1))
-      par(mar = c(0, 6.1, 2.1, 4.1))
-      suppressWarnings(par(mgp = c(3, -0.8, 0)))
-      plot.new()
-      plot.window(ylim = c(0, 1), xlim = range(levels), xaxs = "i", yaxs = "i")
-      rect(levels[-length(levels)],0, levels[-1L],0.3, col = col, border = NA)
-      axis(3,tick = FALSE, at = do.breaks(c(start.point, end.point), end.point/10))
-      title(plot.title)
-      par(mar = c(5.1,4.1,1.1,2.1))
-      suppressWarnings(par(mgp = c(2,1,0)))
-      plot.new()
-      mat.tmp <- t(mat.tmp)
-      x.2D <- (1:dim(mat.tmp)[1] - 1) / (dim(mat.tmp)[1] - 1)
-      y.2D <- (1:dim(mat.tmp)[2] - 1) / (dim(mat.tmp)[2] - 1)
-      plot.window(asp = NA, xlim = range(x.2D), ylim = range(y.2D), "", xaxs = "i", yaxs = "i")
-      .filled.contour(x.2D, y.2D, z = mat.tmp, levels, col = col)
-      box()
-      mtext(drug.col, 1, cex = 1, padj = 3)
-      mtext(drug.row, 2, cex = 1, las = 3, padj = -3)
-      if(!is.null(row.range)) {
-        yconc <- round(row.conc[row.range[1]:row.range[2]], 3)
-      } else {
-        yconc <- round(row.conc, 3)
-      }
-      
-      if(!is.null(col.range)) {
-        xconc <- round(col.conc[col.range[1]:col.range[2]], 3)
-      } else {
-        xconc <- round(col.conc, 3)
-      }
-      
-      axis(side = 1, at = seq(0, 1, by = 1/(length(xconc) - 1)), labels = xconc)
-      axis(side = 2, at = seq(0, 1, by = 1/(length(yconc) - 1)), labels = yconc)
-      print(syn.3d.plot, position = c(0.5,0, 1, 1), newpage = FALSE)
-      fig <- recordPlot()
-    }
-    plots[[i]] <- fig
-    if(save.file) {
-      if (type %in% c("2D", "3D")) {
-        pdf(file.name, width = 10, height = 10)
-      } else {
-        pdf(file.name, width = 12, height = 6)
-      }
-      print(fig)
-      dev.off()
-    }
+# Returns nested associative lists with all data needed for plots of any type.
+.GetPlotData <- function(
+  data, index, legend.start, legend.end, row.range, col.range, len) {
```

Comment:
In the main function, it's y.range and x.range. Do you think we should keep them consistent?

Reply:
Internally, I think row/col is easier to reason about.  When you see this little helper function all by itself, you can read its purpose just like reading English.  For me, the most important thing about the names used to define this function is to make it easy for the reader to understand this function (and its tests).

I left the y/x in the main function because I didn't want to touch the externally-visible interface.  The whole point is that this is supposed to work just like it did before, but with the private internal implementation details rearranged.

So in the end I don't think there's a big concern here.  However that's *my* aesthetic sensibility.  You will own this for the long term, not me, so in the end I'll put it however you want it.